web/client/utils/CatalogUtils.js

```diff
@@ -64,6 +64,63 @@ const getThumb = (dc) => {
     }));
 };
 
+// Extract the relevant information from the wms URL for (RNDT / INSPIRE)
+const extractWMSParamsFromURL = wms => {
+    const params = new URLSearchParams(wms.value);
+    const lowerCaseParams = new URLSearchParams();
+    for (const [name, value] of params) {
+        lowerCaseParams.append(name.toLocaleLowerCase(), value);
+    }
+    const layerName = lowerCaseParams.get('layers');
+    const wmsVersion = lowerCaseParams.get('version');
+    if (layerName) {
+        return {
+            ...wms,
+            protocol: 'OGC:WMS',
+            name: layerName,
+            value: `${wms.value.match( /[^\?]+[\?]+/g)}SERIVCE=WMS${wmsVersion && `&VERSION=${wmsVersion}`}`
+        };
+    }
+    return false;
+};
+
+const getMetaDataDownloadFormat = (protocol = '', value = '') => {
+    let format = '';
+    const formatsMap = [
+        {
+            protocol: /www-download/g,
+            value: /shp\/.*\.zip/g,
+            displayValue: 'shapefile'
+        },
+        {
+            protocol: /www-download/g,
+            value: /grid\/.*\.zip/g,
+            displayValue: 'grid'
+        },
+        {
+            value: /wms\?/g,
+            displayValue: 'WMS'
+        },
+        {
+            protocol: /TR\/xlink/g,
+            value: /kml\/.*\.kmz/g,
+            displayValue: 'KML file'
+        }
+    ];
+    format = formatsMap.filter(formatItem => (formatItem.protocol ? protocol.match(formatItem.protocol) : true) && value.match(formatItem.value))[0]?.displayValue || protocol;
+    return format;
+};
+
+const getURILinkNames = (metadata, locales, uriItem) => {
```

Comment:
It returns links, not only link names. So getURILinks is a better name. 
```suggestion
const getURILinks= (metadata, locales, uriItem) => {
```